superset/result_set.py

```diff
@@ -97,7 +98,7 @@ def __init__(
 
         # put data in a structured array so we can efficiently access each column.
         # cast `data` as list due to MySQL (others?) wrapping results with a tuple.
-        array = np.array(list(data), dtype=numpy_dtype)
+        array = np.array([tuple(row) for row in data], dtype=numpy_dtype)
```

Comment:
Looking at this again, can iteration over the data be avoided here? This could potentially be very expensive for large datasets.

Reply:
We talked about that with @dpgaspar , the problem here is that numpy needs to have this materialized, i.e. generators are off the table. I think we need to revisit this if this becomes a problem, as I believe it should be possible to skip numpy completely. But I would prefer not to have to open up that can of worms right now.